testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3215,6 +3215,260 @@ def test_replication_push_replication_guest_enabled(params_from_base_test_setup,
     replicator.stop(repl)
 
 
+@pytest.mark.listener
+@pytest.mark.replication
+def test_doc_removal_from_channel(params_from_base_test_setup):
+    """
+        @summary:
+        1. Create 2 docs in CBL with channel A, B
+        2. Create user in SGW with channel A, B.
+        3. push_pull replicate to SGW
+        4. remove doc A from channel A
+        5. Remove doc B from channel A , B
+        6. continue push_pull replication
+        7. Verify user can only access doc A, but not doc B
+
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    username = "autotest"
+    password = "password"
+    documentObj = Document(base_url)
```

Comment:
variable name should follow lowercase with underscore

Reply:
done